app/controllers/work_packages/activities_tab_controller.rb

```diff
@@ -51,9 +51,12 @@ def index
   end
 
   def update_streams
-    set_last_server_timestamp_to_headers
-
-    perform_update_streams_from_last_update_timestamp
+    begin
+      set_last_server_timestamp_to_headers
+      perform_update_streams_from_last_update_timestamp
+    rescue StandardError => e
```

Comment:
I am pretty sure that it's impossible to handle the error described in that bug on the backend side. One example of the error (the one in a screenshot) was that the reverse proxy in front of the Rails application returned an HTTP 502.

This means the response of the backend was invisible to the browser, because it just saw the proxy's response.

**tl;dr:** The frontend code must gracefully handle this, not the backend code.

Reply:
I am not hot into the topic (probably another place that @oliverguenther would have more background knowledge, though he is on vacation): There is a clear comment on why we render for errors as well, because we want to render primer error banners as well. So there are expected errors and we want to handle them and render them.

However, I think what the code you linked doesn't do right now is to honor _unexpected_ errors. They should be recognizable. E.g. regular responses indicate the content type as `text/vnd.turbo-stream.html`, I'd expect that unforeseen answers would not do that.

Reply:
P.S.: Thanks for finding the corresponding code. That was very helpful to me in understanding whether we are to blame or Rails is... we are :sweat_smile: